### ISSUE TITLE
Sometimes requeueing doesn't work

### ISSUE THREAD

**Issue by AmosC** [GlobalX/SbManager#49]
In the RequeueAndRemove::RequeueOne() method, the messages are iterated through to find the one to requeue with a matching messageId.  However, it uses the QueueClient::ReceiveBatch() method which may return fewer messages than the number requested, consequently it can happen that the message to requeue is not found in the returned list and so nothing happens.

**Comment by lukeschafer** [comment]
Yeah this is a painful side effect of that ReceiveBatch idiom. I couldn't think of any reasonable way to work around it, except maybe brute-force retries. Maybe there's something there where you can use the messageid to retrieve it explicitely?

**Comment by lukeschafer** [comment]
Also, someone should label these issues

**Comment by drewfreyling** [comment]
@lukeschafer it doesn't look like brute forcing seems to work either :(
